File: vue-app/src/classes/Treatment.ts
import { ITreatment } from "@/interfaces/ITreatment";

export class Treatment implements ITreatment {
  id: number;
  note: string;
  noteForNextTreatment: string;
  horseId: number;
  createdAt: Date;
  updatedAt: Date;
  fileKeysString: string;

  constructor() {
    this.id = 0;
    this.note = "";
    this.noteForNextTreatment = "";
    this.horseId = 0;
    this.createdAt = new Date();
    this.updatedAt = new Date();
    this.fileKeysString = "";
  }
}

File: WebApi/Models/Treatment.cs
public class Treatment : IModel
{
    public int Id { get; set; }
    public string Note { get; set; } = string.Empty;
    public string NoteForNextTreatment { get; set; } = string.Empty;
    public int? HorseId { get; set; }
    public Horse? Horse { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;
    public string FileKeysString { get; set; } = string.Empty;

}
